core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1881,7 +1884,11 @@ class ReplicaManager(val config: KafkaConfig,
     logManager.handleLogDirFailure(dir)
 
     if (sendZkNotification)
-      zkClient.propagateLogDirEvent(localBrokerId)
+      if (zkClient == null) {
```

Comment:
Can we make `zkClient` an `Option`? By the way, we are really close to being able to remove `KafkaZkClient` as a dependency, but I guess we will first need #9577.